aipmiddleware/validateparent_test.go

```diff
@@ -0,0 +1,166 @@
+				response, err := client.ListSites(ctx, &examplefreightv1.ListSitesRequest{
+					Parent: "shippers/1",
+				})
+				assert.DeepEqual(t, server.listSitesResponse, response, protocmp.Transform())
+				assert.NilError(t, err)
+			},
+		},
+
+		{
+			name: "valid wildcard parent response",
+			server: mockFreightServer{
+				listSitesResponse: &examplefreightv1.ListSitesResponse{
+					Sites: []*examplefreightv1.Site{
+						{Name: "shippers/1/sites/1"},
+						{Name: "shippers/1/sites/2"},
+						{Name: "shippers/1/sites/3"},
+					},
+				},
+			},
+			f: func(t *testing.T, ctx context.Context, server *mockFreightServer, conn *grpc.ClientConn) {
+				client := examplefreightv1.NewFreightServiceClient(conn)
+				response, err := client.ListSites(ctx, &examplefreightv1.ListSitesRequest{
+					Parent: "shippers/-",
+				})
+				assert.DeepEqual(t, server.listSitesResponse, response, protocmp.Transform())
+				assert.NilError(t, err)
+			},
+		},
+
+		{
+			name: "invalid parent response",
+			server: mockFreightServer{
+				listSitesResponse: &examplefreightv1.ListSitesResponse{
+					Sites: []*examplefreightv1.Site{
+						{Name: "shippers/1/sites/1"},
+						{Name: "shippers/1/sites/2"},
+						{Name: "shippers/2/sites/3"}, // invalid
+					},
+				},
+			},
+			f: func(t *testing.T, ctx context.Context, server *mockFreightServer, conn *grpc.ClientConn) {
+				client := examplefreightv1.NewFreightServiceClient(conn)
+				response, err := client.ListSites(ctx, &examplefreightv1.ListSitesRequest{
+					Parent: "shippers/1",
+				})
+				assert.Assert(t, response == nil)
+				assert.Equal(t, codes.Internal, status.Code(err))
+			},
+		},
+	} {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			lis, err := net.Listen("tcp", "localhost:0")
+			assert.NilError(t, err)
+			parentValidator := ParentValidator{
+				LogHook: func(info *grpc.UnaryServerInfo, request, response proto.Message, parent, name string) {
+					t.Log(
+						info.FullMethod,
+						prototext.MarshalOptions{}.Format(request),
+						prototext.MarshalOptions{}.Format(response),
+						parent,
+						name,
+					)
+				},
+			}
+			grpcServer := grpc.NewServer(grpc.UnaryInterceptor(parentValidator.UnaryServerInterceptor))
+			examplefreightv1.RegisterFreightServiceServer(grpcServer, &tt.server)
+			var wg sync.WaitGroup
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				assert.NilError(t, grpcServer.Serve(lis))
+			}()
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				<-ctx.Done()
+				grpcServer.GracefulStop()
+			}()
+			t.Cleanup(wg.Wait)
+			conn, err := grpc.DialContext(ctx, lis.Addr().String(), grpc.WithInsecure())
+			assert.NilError(t, err)
+			tt.f(t, ctx, &tt.server, conn)
+		})
+	}
+}
+
+type mockFreightServer struct {
+	examplefreightv1.UnimplementedFreightServiceServer
+	// GetShipper
+	getShipperResponse *examplefreightv1.Shipper
+	getShipperError    error
```

Comment:
either add tests using error, or remove from struct and always return `nil`.

Reply:
Good point - added more tests!